### PR TITLE
Add Phishing Detection Pixels

### PR DESCRIPTION
**Pull request by not-a-rootkit**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1204023833050360/1207976613228507/f
iOS PR: -
macOS PR: -
What kind of version bump will this require?: Major/Minor/Patch

**Description**:
Implements phishing detection pixels as per [✓ Map out pixels and create privacy triage: 0.5 days](https://app.asana.com/0/1204023833050360/1207699541075655/f) and [✓ Privacy Triage: Phishing Detection Pixels on macOS](https://app.asana.com/0/69071770703008/1207738252568650/f)

**Steps to test this PR**:
1. None :/ 

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by not-a-rootkit** [comment]
This will require some client-side changes to pass in the eventMapping, i.e.:

```
        self.detector = detector ?? PhishingDetector(apiClient: detectionClient, dataStore: self.dataStore, eventMapping:
                                                        EventMapping<PhishingDetectionEvents> {event, _, params, _ in
            switch event {
            case .errorPageShown(clientSideHit: let clientSideHit):
                PixelKit.fire(PhishingDetectionEvents.errorPageShown(clientSideHit: clientSideHit))
            case .iframeLoaded:
                PixelKit.fire(PhishingDetectionEvents.iframeLoaded)
            case .visitSite:
                PixelKit.fire(PhishingDetectionEvents.visitSite)
            case .updateTaskFailed48h(error: let error):
                PixelKit.fire(PhishingDetectionEvents.updateTaskFailed48h(error: error))
            }
        })
```

It seems to work and the tests pass.

File: Sources/PhishingDetection/PhishingDetector.swift [hunk 1]
```diff
@@ -55,10 +54,12 @@ public class PhishingDetector: PhishingDetecting {
     let hashPrefixParamLength: Int = 4
 	let apiClient: PhishingDetectionClientProtocol
 	let dataStore: PhishingDetectionDataSaving
+    let eventMapping: EventMapping<PhishingDetectionEvents>
```
Comment (SabrinaTardio) [inline Sources/PhishingDetection/PhishingDetector.swift:57]: seems like these are not aligned

File: Tests/PhishingDetectionTests/Mocks/EventMappingMock.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+public class MockEventMapping: EventMapping<PhishingDetectionEvents> {
+    static var events: [PhishingDetectionEvents] = []
+    static var clientSideHitParam: String?
+    static var errorParam: Error?
+
+    public init() {
+        super.init { event, error, params, _ in
+            Self.events.append(event)
+            switch event {
+            case .errorPageShown(clientSideHit: let clientSideHit):
```
Comment (SabrinaTardio) [inline Tests/PhishingDetectionTests/Mocks/EventMappingMock.swift:32]: can you solve this warning?

**Review by SabrinaTardio — APPROVED** [review]
LGTM please look at the two minor comments and feel free to merge after addressing them

**Pull request merged**